docs/argo-server-sso.md

```diff
@@ -188,6 +188,25 @@ sso:
   userInfoPath: /oauth2/v1/userinfo
 ```
 
+## Filter groups
+
+> v3.5 and after
+
+```yaml
+sso:
+    # Specify a list of regular expressions to filter the groups returned by the OIDC provider.
+    filterGroupsRegex:
+    - ".*argo-wf.*"
+    - ".*argo-workflow.*"
+```
+
+- `filterGroups` is a list of regular expressions that will be used to filter the groups returned by the OIDC provider.
+- This is an optional field. If not specified, no filtering will be done.
+- The regular expressions are case-sensitive.
+- If the list is not empty, only groups that match at least one of the regular expressions will be used in SSO RBAC.
+- Argo workflows uses cookies to store the groups' information. If the number of groups returned by OIDC provider is too large, the cookie size will exceed the  **4KB** limit and the login will fail. To avoid this, you can use `filterGroups` to filter out the groups that are not relevant to SSO RBAC.
+- Also, useful if you have multiple applications using the same OIDC provider, and you want to restrict access to only those groups that are relevant to Argo Workflows.
+
 ### Example Expression
```

Comment:
this is an h3 that is part of "Custom Claims". the new section should either be above "Custom Claims" or at the very bottom. currently it got spliced in between "Custom Claims"'s sections